libs/application/templates/directorate-of-immigration/citizenship/src/fields/Review/ChildrenReview.tsx

```diff
@@ -0,0 +1,60 @@
+import {
+  ApplicantChildCustodyInformation,
+  FieldBaseProps,
+} from '@island.is/application/types'
+import { personal, review, selectChildren } from '../../lib/messages'
+import { Box, GridColumn, GridRow, Text } from '@island.is/island-ui/core'
+import DescriptionText from '../../components/DescriptionText'
+import { useLocale } from '@island.is/localization'
+import { Routes } from '../../lib/constants'
+import SummaryBlock from '../../components/SummaryBlock'
+
+interface ChildrenReviewProps extends FieldBaseProps {
+  selectedChildren: Array<ApplicantChildCustodyInformation> | undefined
+  goToScreen?: (id: string) => void
+  route: Routes
+}
+
+export const ChildrenReview = ({
```

Comment:
I noticed how the review sections are pretty similar. Could you make a generic review field of some sort that would allow you to avoid a lot of repetition?

Something like this, with no guarantee that this code will run.
```
export const GenericReview: FC<GenericReviewProps> = ({
  title,
  descriptionTextProps,
  editAction,
  data,
}) => {
  return (
    <SummaryBlock editAction={editAction}>
      <Box paddingBottom={4}>
        <DescriptionText text={title} textProps={descriptionTextProps} />
        {data.map((rowData, rowIndex) => (
          <GridRow key={rowIndex}>
            {rowData.leftColumn.map((item, columnIndex) => (
              <GridColumn span={'1/2'} key={`left-${columnIndex}`}>
                <Text>
                  {item.label}: {item.value}
                </Text>
              </GridColumn>
            ))}
            {rowData.rightColumn.map((item, columnIndex) => (
              <GridColumn span={'1/2'} key={`right-${columnIndex}`}>
                <Text>
                  {item.label}: {item.value}
                </Text>
              </GridColumn>
            ))}
          </GridRow>
        ))}
      </Box>
    </SummaryBlock>
  )
}
```

Reply:
I created a generic review but I could really only use it in a few places since the reviews are slightly different and the Generic review would really just get so messy